esphome/components/tm1621/tm1621.cpp

```diff
@@ -0,0 +1,283 @@
+#include "tm1621.h"
+#include "esphome/core/log.h"
+#include "esphome/core/helpers.h"
+#include "esphome/core/hal.h"
+
+namespace esphome {
+namespace tm1621 {
+
+static const char *const TAG = "tm1621";
+
+const uint8_t TM1621_PULSE_WIDTH = 10;  // microseconds (Sonoff = 100)
+
+const uint8_t TM1621_SYS_EN = 0x01;     // 0b00000001
+const uint8_t TM1621_LCD_ON = 0x03;     // 0b00000011
+const uint8_t TM1621_TIMER_DIS = 0x04;  // 0b00000100
+const uint8_t TM1621_WDT_DIS = 0x05;    // 0b00000101
+const uint8_t TM1621_TONE_OFF = 0x08;   // 0b00001000
+const uint8_t TM1621_BIAS = 0x29;       // 0b00101001 = LCD 1/3 bias 4 commons option
+const uint8_t TM1621_IRQ_DIS = 0x80;    // 0b100x0xxx
+
+enum Tm1621Device { TM1621_USER, TM1621_POWR316D, TM1621_THR316D };
+
+const uint8_t TM1621_COMMANDS[] = {TM1621_SYS_EN,  TM1621_LCD_ON,   TM1621_BIAS,   TM1621_TIMER_DIS,
+                                   TM1621_WDT_DIS, TM1621_TONE_OFF, TM1621_IRQ_DIS};
+
+const char TM1621_KCHAR[] PROGMEM = {"0|1|2|3|4|5|6|7|8|9|-| "};
+//                                          0     1     2     3     4     5     6     7     8     9     -     off
+const uint8_t TM1621_DIGIT_ROW[2][12] = {{0x5F, 0x50, 0x3D, 0x79, 0x72, 0x6B, 0x6F, 0x51, 0x7F, 0x7B, 0x20, 0x00},
+                                         {0xF5, 0x05, 0xB6, 0x97, 0x47, 0xD3, 0xF3, 0x85, 0xF7, 0xD7, 0x02, 0x00}};
+
+void TM1621Display::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up TM1621...");
+
+  this->cs_pin_->setup();  // OUTPUT
+  this->cs_pin_->digital_write(true);
+  this->data_pin_->setup();  // OUTPUT
+  this->data_pin_->digital_write(true);
+  this->read_pin_->setup();  // OUTPUT
+  this->read_pin_->digital_write(true);
+  this->write_pin_->setup();  // OUTPUT
+  this->write_pin_->digital_write(true);
+
+  this->state_ = 100;
+
+  this->cs_pin_->digital_write(false);
+  delayMicroseconds(80);
+  this->read_pin_->digital_write(false);
+  delayMicroseconds(15);
+  this->write_pin_->digital_write(false);
+  delayMicroseconds(25);
+  this->data_pin_->digital_write(false);
+  delayMicroseconds(TM1621_PULSE_WIDTH);
+  this->data_pin_->digital_write(true);
+
+  for (unsigned char tm1621_command : TM1621_COMMANDS) {
```

Comment:
Please replace all `unsigned char` with `uint8_t`

```suggestion
  for (uint8_t tm1621_command : TM1621_COMMANDS) {
```

Reply:
Done
https://github.com/Philippe12/esphome/commit/fb1161bef4f08d77d59a7eef40bd0e8b3f336973